rust/cardano-blockchain-types/src/fork.rs

```diff
@@ -6,13 +6,40 @@
 //!
 //! Note: This fork terminology is different from fork in blockchain.
 
+use std::fmt;
+
 use crate::conversion::from_saturating;
 
 #[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, PartialOrd)]
 /// Counter that is incremented every time there is a roll-back in live-chain.
 pub struct Fork(u64);
 
 impl Fork {
+    /// Fork for data that read from the blockchain during a backfill on initial sync
+    pub const BACKFILL: Self = Self(1);
+    /// Fork count for the first live block.
+    pub const FIRST_LIVE: Self = Self(2);
+    /// Fork for immutable data. This indicates that there is no roll-back.
+    pub const IMMUTABLE: Self = Self(0);
+
+    /// Is the fork for immutable data.
+    #[must_use]
+    pub fn is_immutable(&self) -> bool {
+        self.0 == 0
```

Comment:
use the constant, not repeating the magic value.
```suggestion
        self == Self::IMMUTABLE
```